R/principalcomponentanalysis.R

```diff
@@ -161,23 +165,34 @@ PrincipalComponentAnalysis <- function(jaspResults, dataset, options, ...) {
 
   if (options$factorMethod == "manual") return(options$numberOfFactors)
 
-  fa <- try(psych::fa.parallel(dataset, plot = FALSE, fa = options$parallelMethod))
-  if (isTryError(fa)) return(1)
+  if (options[["basedOn"]] == "mixedCorrelationMatrix") {
+    polyTetraCor <- psych::mixedCor(dataset)
+    parallelResult <- try(psych::fa.parallel(polyTetraCor$rho,
+                                 plot = FALSE,
+                                 fa = options$parallelMethod,
+                                 n.obs = nrow(dataset)))
+  }
+  else {
+    parallelResult <- try(psych::fa.parallel(dataset, plot = FALSE, fa = options$parallelMethod))
+  }
+
+  if (isTryError(parallelResult)) return(1)
```

Comment:
so am I getting it right that if `fa.parallel` fails we just silently set the number of components to 1? Should we display an error message perhaps?